File: 504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        # time: O(n), space: O(n)
        isNeg = False
        res = ""
        if num == 0:
            return "0"
        elif num < 0:
            isNeg = True
            num = abs(num)
        while num:
            res = str(num % 7) + res
            num //= 7
        return "-" + res if isNeg else res

File: 504.java
class Solution {
    public String convertToBase7(int num) {
        // time: O(logn), space: O(1)
        if (num == 0) return "0";
        StringBuilder str = new StringBuilder();
        boolean neg = false;
        if (num < 0) neg = true;
        num = Math.abs(num);
        while (num > 0) {
            str.append(num % 7);
            num /= 7;
        }
        if (neg) str.append("-");
        return str.reverse().toString();
    }
}